avae/vis.py

```diff
@@ -531,25 +537,62 @@ def recon_plot(img, rec, name="trn"):
     fname_in = "plots/" + str(name) + "_recon_in.png"
     fname_out = "plots/" + str(name) + "_recon_out.png"
 
-    img = img[:, :, :, :, img.shape[-1] // 2]
-    rec = rec[:, :, :, :, img.shape[-1] // 2]
+    img_2d = img[:, :, :, :, img.shape[-1] // 2]
+    rec_2d = rec[:, :, :, :, img.shape[-1] // 2]
 
     plt.subplots(figsize=(10, 10))
-    img = torchvision.utils.make_grid(img.cpu(), 10, 2).numpy()
-    plt.imshow(np.transpose(img, (1, 2, 0)))  # channels last
+    img_2d = torchvision.utils.make_grid(img_2d.cpu(), 10, 2).numpy()
+    plt.imshow(np.transpose(img_2d, (1, 2, 0)))  # channels last
     if not os.path.exists("plots"):
         os.mkdir("plots")
     plt.savefig(fname_in)
     plt.close()
 
     plt.subplots(figsize=(10, 10))
-    rec = torchvision.utils.make_grid(rec.detach().cpu(), 10, 2).numpy()
-    plt.imshow(np.transpose(rec, (1, 2, 0)))  # channels last
+    rec_2d = torchvision.utils.make_grid(rec_2d.detach().cpu(), 10, 2).numpy()
+    plt.imshow(np.transpose(rec_2d, (1, 2, 0)))  # channels last
```

Comment:
we might have to do some dynamic exposure adjustment to make sure we can visualise the reconstructions. E.g:
Default: 
![trn_recon_out](https://github.com/alan-turing-institute/affinity-vae/assets/11162074/54c39334-4092-4d65-8940-bff65b47d939)
Exposure corrected:
![trn_recon_out](https://github.com/alan-turing-institute/affinity-vae/assets/11162074/d1f3f0c2-b22f-4390-8be9-ce8dc215e210)

Reply:
this will be sorted in a new PR/